Source/MQTTnet/Server/MqttClientSessionsManager.cs

```diff
@@ -425,10 +426,13 @@ MqttClientConnection CreateClientConnection(MqttConnectPacket connectPacket, Mqt
                     _connections[connectPacket.ClientId] = connection;
                 }
 
-                existingConnection?.StopAsync(MqttClientDisconnectReason.SessionTakenOver).GetAwaiter().GetResult();
-
-                return connection;
+                if (existingConnection != null)
+                {
+                    await existingConnection.StopAsync(MqttClientDisconnectReason.SessionTakenOver).ConfigureAwait(false);
```

Comment:
Changes in this file is not required, but `Session_Tests.Manage_Session_MaxParallel()` may failed, because too many threads are blocked by `GetResult()` here when change https://github.com/chkr1011/MQTTnet/blob/7eb6d2d1e3d0480b89cbdafe0dc4380ea5dd055c/Tests/MQTTnet.Core.Tests/Session_Tests.cs#L101 from 10 to 50.